nodes/obstacle_visualizer.py

```diff
@@ -0,0 +1,139 @@
+            obstacle_name = "~obstacle" + str(obstacle_idx)
+            self._obstacles_infos.update(
+                {obstacle_name: rospy.get_param(obstacle_name)}
+            )
+            obstacle_idx += 1
+
+        sdf_generator = SDFGenerator()
+        self._spawn_model_requests = []
+        self._markers = MarkerArray()
+        header = Header(frame_id="world", stamp=rospy.Time.now())
+        for iter, (key, obstacle) in enumerate(self._obstacles_infos.items()):
+            
+            pose=self._parse_poses(obstacle)
+ 
+            sp_req = SpawnModelRequest(
+                model_name = key[1:],
+                robot_namespace = "",
+                initial_pose = pose,
+                reference_frame = header.frame_id
+            )
+            
+
+            m = Marker(
+                header=header,
+                ns="",
+                id=iter,
+                action=Marker.MODIFY,
+                pose = pose,
+                color=ColorRGBA(r=0.58, g=0.0, b=0.98, a=0.7),
+                lifetime=rospy.Duration(0.1),
+                frame_locked=False,
+            )
+            if obstacle["type"] == "sphere":
+                m.scale = Vector3(**dict(zip("xyz", [obstacle["radius"] * 2] * 3)))
+                m.type = Marker.SPHERE
+                sp_req.model_xml = sdf_generator.generate_sphere(key[1:], obstacle["radius"])
+            if obstacle["type"] == "cylinder":
+                m.scale = Vector3(x = obstacle["radius"], y = obstacle["radius"], z = obstacle["halfLength"])
+                m.type = Marker.CYLINDER
+                sp_req.model_xml = sdf_generator.generate_cylinder(key[1:], obstacle["radius"],obstacle["halfLength"] )
+            if obstacle["type"] == "box":
+                m.scale = Vector3(x = obstacle["x"], y = obstacle["y"], z = obstacle["z"])
+                m.type = Marker.CUBE
+                sp_req.model_xml = sdf_generator.generate_box(key[1:], [obstacle["x"],obstacle["y"], obstacle["z"] ])
+            if obstacle["type"] == "capsule":
+                m.scale = Vector3(x = obstacle["radius"], y = obstacle["radius"], z = obstacle["halfLength"])
+                m.type = Marker.CYLINDER
+                sp_req.model_xml = sdf_generator.generate_capsule(key[1:], obstacle["radius"],obstacle["halfLength"] )              
+            print(f"key: {key[1:]}")
+            print(f"sp_req.model_xml: {sp_req.model_xml}")
+            self._spawn_model_requests.append(sp_req)
+            self._markers.markers.append(m)
+
+
+        rospy.wait_for_service("/gazebo/spawn_urdf_model")
+        try:
+            model_spawner = rospy.ServiceProxy("/gazebo/spawn_sdf_model", SpawnModel)
+            for sp_req in self._spawn_model_requests:
+                resp = model_spawner(sp_req)
+                if not resp.success:
+                    rospy.logerr("spawning model didn't work.")
+        except rospy.ServiceException as e:
+            print("Service call failed: %s" % e)
```

Comment:
Add ROS paramter disabling this part of the code so it can be later used on real robot